plugins/mmstd_datatools/src/table/MMFTDataSource.cpp

```diff
@@ -1,145 +1,179 @@
+    MakeSlotAvailable(&getDataSlot_);
 }
 
-MMFTDataSource::~MMFTDataSource(void) {
-    this->Release();
+MMFTDataSource::~MMFTDataSource() {
+    Release();
 }
 
-bool MMFTDataSource::create(void) {
+bool MMFTDataSource::create() {
     // nothing to do
     return true;
 }
 
-void MMFTDataSource::release(void) {
-    this->columns.clear();
-    this->values.clear();
+void MMFTDataSource::release() {
+    columns_.clear();
+    values_.clear();
 }
 
 void MMFTDataSource::assertData(void) {
-    if (!this->filenameSlot.IsDirty()) {
+    if (!filenameSlot_.IsDirty() && !reload_) {
         return; // nothing to do
     }
-    
-    this->filenameSlot.ResetDirty();
 
-    this->columns.clear();
-    this->values.clear();
+    filenameSlot_.ResetDirty();
+    reload_ = false;
 
+    columns_.clear();
+    values_.clear();
 
-    vislib::sys::FastFile file;
-    if (!file.Open(filenameSlot.Param<core::param::FilePathParam>()->Value(), vislib::sys::File::READ_ONLY, vislib::sys::File::SHARE_READ, vislib::sys::File::OPEN_ONLY)) {
-        megamol::core::utility::log::Log::DefaultLog.WriteError("Unable to open file \"%s\". Abort.", vislib::StringA(filenameSlot.Param<core::param::FilePathParam>()->Value()).PeekBuffer());
+    std::string filename(filenameSlot_.Param<core::param::FilePathParam>()->Value().PeekBuffer());
+    std::ifstream file(filename, std::ios::binary);
+    if (!file.is_open()) {
+        megamol::core::utility::log::Log::DefaultLog.WriteError("Unable to open file \"%s\". Abort.", filename.c_str());
         return;
     }
 
-#define ABORT_ERROR(...) { \
-        megamol::core::utility::log::Log::DefaultLog.WriteError(__VA_ARGS__); \
-        file.Close(); \
-        this->columns.clear(); \
-        this->values.clear(); \
-        return; \
-    }
-#define ASSERT_READ(A, S) if (file.Read((A), (S)) != (S)) ABORT_ERROR("Read error %d", __LINE__)
-
-    vislib::StringA magicID;
-    ASSERT_READ(magicID.AllocateBuffer(6), 6);
-    if (!magicID.Equals("MMFTD")) ABORT_ERROR("Wrong file format magic ID");
-    uint16_t version;
-    ASSERT_READ(&version, 2);
-    if (version != 0) ABORT_ERROR("Wrong file format version number");
-
-    uint32_t colCnt;
-    ASSERT_READ(&colCnt, 4);
-    columns.resize(colCnt);
-
-    for (uint32_t c = 0; c < colCnt; ++c) {
-        TableDataCall::ColumnInfo& ci = columns[c];
-        uint16_t nameLen;
-        ASSERT_READ(&nameLen, 2);
-        vislib::StringA name;
-        ASSERT_READ(name.AllocateBuffer(nameLen), nameLen);
-        ci.SetName(name.PeekBuffer());
-        uint8_t type;
-        ASSERT_READ(&type, 1);
-        ci.SetType(
-            (type == 1) ? TableDataCall::ColumnType::CATEGORICAL
-            : TableDataCall::ColumnType::QUANTITATIVE);
-        float f;
-        ASSERT_READ(&f, 4);
-        ci.SetMinimumValue(f);
-        ASSERT_READ(&f, 4);
-        ci.SetMaximumValue(f);
-    }
+    try {
+        std::string magicId = read_string(file, 6);
+        if (magicId != "MMFTD") {
```

Comment:
Will be removed in C++20. Use `compare` instead.
Should also be able to cope with following `\0` -> no need to delete them.